test/Microsoft.OpenApi.Readers.Tests/V3Tests/OpenApiDocumentTests.cs

```diff
@@ -1204,5 +1204,15 @@ public void GlobalSecurityRequirementShouldReferenceSecurityScheme()
                 Assert.Same(securityRequirement.Keys.First(), openApiDoc.Components.SecuritySchemes.First().Value);
             }
         }
+
+        [Fact]
+        public void HeaderParameterShouldAllowExample()
+        {
+            using (var stream = Resources.GetStream(Path.Combine(SampleFolderPath, "apiWithFullHeaderComponent.yaml")))
+            {
+                var openApiDoc = new OpenApiStreamReader().Read(stream, out var diagnostic);
+                Assert.Equal(0, diagnostic.Errors.Count);
```

Comment:
The test is a bit too loose. I think it will pass even without adding the new properties in the deserializer (those properties will simply be ignored.)

Could you build the object and compare it against your object "openApiDoc"? See the ParseModifiedPetStoreDocumentWithTagAndSecurityShouldSucceed test for example.